docs/translations/translation-loading.md

```diff
@@ -0,0 +1,331 @@
+"_nx_noop:3c,1,2;__ngettext_noop:1,2\n"
+"X-Poedit-SearchPath-0: .\n"
+"X-Poedit-SearchPathExcluded-0: *.min.js\n"
+"X-Poedit-SearchPathExcluded-1: vendor\n"
+"X-Poedit-SearchPathExcluded-2: node_modules\n"
+
+#: assets/js/atomic/blocks/product-elements/add-to-cart/constants.js:8
+msgid "Add to Cart"
+msgstr ""
+
+#: assets/js/atomic/blocks/product-elements/add-to-cart/edit.js:39
+#: assets/js/blocks/handpicked-products/block.js:42
+#: assets/js/blocks/product-best-sellers/block.js:34
+#: assets/js/blocks/product-category/block.js:157
+#: assets/js/blocks/product-new/block.js:36
+#: assets/js/blocks/product-on-sale/block.js:52
+#: assets/js/blocks/product-tag/block.js:121
+#: assets/js/blocks/product-top-rated/block.js:36
+#: assets/js/blocks/products-by-attribute/block.js:46
+#: assets/js/blocks/single-product/edit/layout-editor.js:56
+msgid "Layout"
+msgstr ""
+
+[...]
+```
+
+See also <https://developer.wordpress.org/plugins/internationalization/localization/#pot-portable-object-template-files>.
+
+### PO (Portable Object) file
+
+PO stands for `Portable Object`, contains both the original and the translated strings and is based on the POT file. Similar to the POT file, it can also be created manually using [WP-CLI](https://wp-cli.org/) or [Poedit](https://poedit.net/). As mentioned before, this step is not necessary for the WooCommerce Blocks plugin, as it's hosted on WordPress.org. The PO file will be generated within 30 minutes after a new translation had been added via <https://translate.wordpress.org/projects/wp-plugins/woo-gutenberg-products-block/>.
+
+The PO file is human-readable and named `woo-gutenberg-products-block-{LANGUAGE-CODE}.po`, e.g. `woo-gutenberg-products-block-de_DE.po`. It can be found in the `/wp-content/languages/plugin` folder. The PO file looks like this:
+
+```po
+# Translation of Plugins - WooCommerce Blocks - Stable (latest release) in German
+# This file is distributed under the same license as the Plugins - WooCommerce Blocks - Stable (latest release) package.
+msgid ""
+msgstr ""
+"PO-Revision-Date: 2022-05-22 10:58:25+0000\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Plural-Forms: nplurals=2; plural=n != 1;\n"
+"X-Generator: GlotPress/4.0.0-alpha.1\n"
+"Language: de\n"
+"Project-Id-Version: Plugins - WooCommerce Blocks - Stable (latest release)\n"
+
+#: assets/js/blocks/featured-product/block.json
+#: build/featured-product/block.json
+msgctxt "block description"
+msgid "Visually highlight a product or variation and encourage prompt action."
+msgstr "Ein Produkt oder eine Variante visuell hervorheben und zum sofortigen Handeln auffordern."
+
+#: assets/js/blocks/featured-product/block.json
+#: build/featured-product/block.json
+msgctxt "block title"
+msgid "Featured Product"
+msgstr "Hervorgehobenes Produkt"
+
+[...]
+```
+
+See also <https://developer.wordpress.org/plugins/internationalization/localization/#po-portable-object-files>.
```

Comment:
Not sure if this is worth noting, but we we create the `po` and `mo` files manually from the pot file above, then putting those file in `plugin-root-dir/languages/` won't work because we don't load the text domain using `load_plugin_textdomain()`.

This can be confused because the JSON translations in `plugin-root-dir/languages/`, in the opposite, are loaded and have a higher priority than ones in `wp-content/languages/plugins/`.

Reply:
Based on https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/#plugins-on-wordpress-org, I understood that since WordPress 4.6 `load_plugin_textdomain()` is no longer required if the plugin is hosted on WordPress.org and the plugin slug and the text domain are equal. Is that information incorrect?

Reply:
> since WordPress 4.6 `load_plugin_textdomain()` is no longer required if the plugin is hosted on WordPress.org and the plugin slug and the text domain are equal. Is that information incorrect?

@nielslange Yes. That's correct.

I was talking about the case when developers manually create `pot`, then `po` and `mo` files locally. When working with translation issues, I created PO and MO files manually and put them in the `plugin-dir/languages`, which won't work because we don't load po/mo files from the `plugin-dir/languages`. It may be only I encountered that issue so it's okay to keep this file as it is now.

Reply:
Just want to clarify my point:

Without `load_plugin_textdomain` in the plugin, only PO and MO files from `wp-content/languages/plugins` will be loaded. Putting those files in `plugin-dir/languages` won't work.

This is never an issue if translations are downloaded automatically. But it may be an issue if developers create them manually.